File: app/src/main/java/net/onamap/android/util/AssetUtils.java
package net.onamap.android.util;

import android.content.Context;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class AssetUtils {
    public static String doit(Context context, String filename) {
        String str = null;

        try {
            StringBuilder buf = new StringBuilder();
            InputStream json = null;
            json = context.getAssets().open(filename);

            BufferedReader in =
                    new BufferedReader(new InputStreamReader(json, "UTF-8"));

            while ((str = in.readLine()) != null) {
                buf.append(str);
            }

            in.close();
            str = buf.toString();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return str;
    }
}

File: app/src/main/java/net/onamap/android/util/AssetUtils.kt
package net.onamap.android.util

import android.content.Context
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader

object AssetUtils {
    fun doit(context: Context, filename: String?): String? {
        var str: String? = null
        try {
            val buf = StringBuilder()
            var json: InputStream? = null
            json = context.assets.open(filename!!)
            val inputStream = BufferedReader(InputStreamReader(json, "UTF-8"))
            while (inputStream.readLine().also { str = it } != null) {
                buf.append(str)
            }
            inputStream.close()
            str = buf.toString()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return str
    }
}